### PR TITLE
nixd: add a json schema

### LINKED ISSUE #222: [feature] provide a json schema for .nixd.json
Why not provide a json schema to <https://www.schemastore.org/json/> to provide language server support for `.nixd.json`? It can make users to edit `.nixd.json` easier. TIA!

### PR DESCRIPTION
**Pull request by Freed-Wu**
Fix #222

1. Create a json schema
2. Add its URL to schemastore.

Now the json schema is added to this repo. I'll send a PR to add the URL of this json schema to [schema store](https://github.com/SchemaStore/schemastore)

### CODE REVIEW

**Comment by Freed-Wu** [comment]
A `.nixd.json` for test:

```json
{
  "options": {
    "enable": 1,
    "target": {
      "args": [
        "--expr",
        "(import <nixpkgs/nixos> { configuration = ./configuration.nix; }).options"
      ],
      "installable": ""
    }
  }
}
```

```python
import json
import os
from jsonschema import validate

with open(os.path.expanduser("nixd-schema.json"), "rb") as f:
    schema = json.load(f)
with open(os.path.expanduser(".nixd.json"), "rb") as f:
    data = json.load(f)
validate(data, schema)
```

```text
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/flnhq1gqri84ws4lv6n28y5mfg87xnz5-python3-3.11.7-env/lib/python3.11/site-packages/jsonschema/validators.py", line 1308, in validate
    raise error
jsonschema.exceptions.ValidationError: 1 is not of type 'boolean'
```

**Review by inclyc — APPROVED** [review]
Thanks! Do we need any commits on our markdown documentation?

**Comment by Freed-Wu** [comment]
> Do we need any commits on our markdown documentation?

Added some screenshots to documentation.

File: nixd/docs/user-guide.md [hunk 1]
```diff
@@ -64,6 +64,17 @@ nix build -L .#
 
 ### Configuration
 
+`.nixd.json`, the configuration of `nixd`, supports [json schema](https://json-schema.org/).
```
Comment (inclyc) [inline nixd/docs/user-guide.md:67]: Maybe better write these paragraphs after the description.

File: nixd/docs/user-guide.md [hunk 2]
```diff
@@ -64,6 +64,17 @@ nix build -L .#
...
+So if your editor supports
+[LSP](https://microsoft.github.io/language-server-protocol/implementors/servers/),
+you can get completions, diagnostics and more:
+
+![completion](https://github.com/nix-community/nixd/assets/32936898/013367c6-63d8-4bba-9057-c5701c2f36c1)
+
+![diagnostic](https://github.com/nix-community/nixd/assets/32936898/d285eb69-f023-4573-a8e5-8d5501ae3d16)
+
+The pictures are [neovim](https://neovim.org/) with the plugin
```
Comment (inclyc) [inline nixd/docs/user-guide.md:76]: Maybe just remove this line or use footnotes.

Comment (Freed-Wu) [reply nixd/docs/user-guide.md]: Done.

File: nixd/docs/user-guide.md (hunk 1 above)
Comment (Freed-Wu) [reply nixd/docs/user-guide.md]: Done.

File: nixd/docs/user-guide.md [hunk 3]
```diff
@@ -148,6 +148,18 @@ Typically, you can write a nix file, and evaluate the result into `.nixd.json`,
...
+`.nixd.json`, the configuration of `nixd`, supports [json schema](https://json-schema.org/).
+So if your editor supports
+[LSP](https://microsoft.github.io/language-server-protocol/implementors/servers/),
+you can get completions, diagnostics and more[^json-schema]:
+
+![completion](https://github.com/nix-community/nixd/assets/32936898/013367c6-63d8-4bba-9057-c5701c2f36c1)
+
+![diagnostic](https://github.com/nix-community/nixd/assets/32936898/d285eb69-f023-4573-a8e5-8d5501ae3d16)
+
+[^json-schema]: The pictures are [neovim](https://neovim.org/) with the plugin
```
Comment (inclyc) [inline nixd/docs/user-guide.md:160]:
```diff
- The pictures are [neovim](https://neovim.org/) with the plugin
+ These pictures were captured in [neovim](https://neovim.org/) with the plugin
```

Comment (Freed-Wu) [reply nixd/docs/user-guide.md]: Done. Really be better English.

**Pull request merged**